x-pack/solutions/security/plugins/security_solution/server/endpoint/services/actions/clients/microsoft/defender/endpoint/ms_defender_endpoint_actions_client.ts

```diff
@@ -378,6 +380,71 @@ export class MicrosoftDefenderEndpointActionsClient extends ResponseActionsClien
     return msDefenderEndpointGetMachineDetailsApiResponse;
   }
 
+  /**
+   * Fetches and validates the details of a specific action from Microsoft Defender for Endpoint.
+   * This method ensures that the action returned by MDE matches our expected script name and action ID,
+   * detecting when MDE throttles/replaces our action with an existing one.
+   *
+   * @param machineActionId - The Microsoft Defender machine action ID returned from sendAction
+   * @param expectedScriptName - The script name we requested to run
+   * @param expectedActionId - The Kibana action ID we included in the comment
+   * @returns Validation result with isValid flag, action details, and error message if validation fails
+   * @internal
+   */
+  private async fetchAndValidateActionDetails(
+    machineActionId: string,
+    expectedScriptName: string,
+    expectedActionId: string
+  ): Promise<ActionValidationResult> {
+    const fetchAction = async () => {
+      const params: MicrosoftDefenderEndpointGetActionsParams = {
+        id: [machineActionId],
+        pageSize: 1,
+      };
+
+      const response = await this.sendAction<
+        MicrosoftDefenderEndpointGetActionsResponse,
+        MicrosoftDefenderEndpointGetActionsParams
+      >(MICROSOFT_DEFENDER_ENDPOINT_SUB_ACTION.GET_ACTIONS, params);
+
+      return response.data?.value?.[0];
+    };
+
+    try {
+      this.log.debug(
+        `Fetching action details from MDE API for machineActionId [${machineActionId}]`
+      );
+
+      // Retry once after a short delay if action was not found on first attempt
+      // This handles the case where MDE has slight API indexing lag
+      const actionDetails = await retryWithDelay(fetchAction);
```

Comment:
To reviewers: any suggestions on how to deal with this any other way? 
This is the scenario:
- we run runscript, and get the response with `id` that we save as `machineActionId`. 
- when we try to fetch details for this machineActionId - the list is sometimes empty, but after a retry - we're good. 
- I am using list API instead of details API, just to not introduce a new sub-action in the connector.